lib/fray/src/fray/examples/fake_llm_worker.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2025 The Marin Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example worker that simulates LLM inference with fake responses.
+
+This worker is used for testing the WorkerPool without requiring actual LLM APIs.
+It processes math problems from a task queue and returns mock answers.
+"""
+
+import argparse
+import logging
+import random
+import time
+
+from fray.cluster.local.file_queue import FileQueue
+
+logger = logging.getLogger(__name__)
+
+
+def process_task(task: dict) -> dict:
+    """Process a single task (simulate LLM inference).
+
+    Args:
+        task: Task dict with 'problem' and 'id' fields
+
+    Returns:
+        Result dict with 'problem_id' and 'answer' fields
+    """
+    # Simulate processing time
+    time.sleep(random.uniform(0.1, 0.5))
+
+    # Generate a fake answer
+    problem_id = task["id"]
+    problem = task["problem"]
+
+    # Simple mock answer generation
+    answer = f"The answer to '{problem}' is 42 (simulated)"
+
+    return {
+        "problem_id": problem_id,
+        "answer": answer,
+    }
+
+
+def worker_loop(task_queue_name: str, result_queue_name: str) -> None:
+    """Main worker loop that processes tasks from queue.
+
+    Args:
+        task_queue_name: Name of the task queue to consume from
+        result_queue_name: Name of the result queue to publish to
+    """
+    logger.info(f"Worker starting: task_queue={task_queue_name}, result_queue={result_queue_name}")
+
+    # Create queue connections
+    task_queue = FileQueue(name=task_queue_name)
```

Comment:
queue should come from the cluster.

we should define a current_cluster() using a ContextVar, see current_flow() etc for zephyr, fetch the queue handle from the cluster